src/main/java/org/ladocuploader/app/cli/SubmissionTransfer.java

```diff
@@ -74,35 +74,51 @@ public SubmissionTransfer(SubmissionRepository submissionRepository, Transmissio
     this.ftpsClient = ftpsClient;
   }
 
+  @ShellMethod(key = "transferSubmissions")
   public void transferSubmissions() {
     // Get submissions to transfer
     String batchSeq = Long.toString(transmissionRepository.nextValueBatchSequence());
     String batchIndex = Strings.padStart(batchSeq, BATCH_INDEX_LEN, '0');
+    UUID uuid = UUID.randomUUID();
     String zipFileName = batchIndex + ".zip";
-    log.info(String.format("Beginning transfer of batch %s", batchIndex));
+    log.info(String.format("Beginning transfer of %s: batch %s", uuid, batchIndex));
+
+    // Stats on transfers
+    int successful = 0, failed = 0;
 
     // for each submission, add to zipfile
     List<Submission> submissionsBatch = transmissionRepository.submissionsToTransmit(Sort.unsorted(), TransmissionType.SNAP);
+    int subfolderidx = 1;
     try (FileOutputStream baos = new FileOutputStream(zipFileName);
          ZipOutputStream zos = new ZipOutputStream(baos)) {
       long now = new Date().getTime();
       StringBuilder docMeta = new StringBuilder();
-      int subfolderidx = 1;
       for (Submission submission : submissionsBatch) {
         if (submission.getSubmittedAt().getTime() + TWO_HOURS > now) {
           // Give a 2-hour wait for folks to upload documents
           continue;
         }
 
+        Transmission transmission = transmissionRepository.findBySubmissionAndTransmissionType(submission, TransmissionType.SNAP);
+
         String subfolder = Integer.toString(subfolderidx++);
         try {
           log.info("Generate applicant summary");
           packageSnapApplication(batchIndex, zos, docMeta, submission, subfolder);
 
           log.info("Adding uploaded docs");
           packageUploadedDocuments(batchIndex, zos, docMeta, submission, subfolder);
+
+          transmission.setStatus(TransmissionStatus.Complete);
+          updateTransmission(uuid, transmission);
+          successful++;
         } catch (Exception e) {
           log.error("Error generating file collection for submission ID {}", submission.getId(), e);
+
+          transmission.setDocumentationErrors(Map.of("error", e.getMessage()));
+          transmission.setStatus(TransmissionStatus.Failed);
+          updateTransmission(uuid, transmission);
```

Comment:
@sree-cfa - curious if it's worthwhile including the subfolder details to hone in on what might cause the error.

I realize that the whole transmission would fail but I am curious if we will ever end up with big enough transmissions that it might become difficult to troubleshoot.

Reply:
Oh, yeah good call out - will add that